backend/models/job.js

```diff
@@ -18,155 +18,205 @@
+	return db.getCollection(teamspace, "jobs");
+}
+
+const Job = {};
+
+Job.addDefaultJobs = function(teamspace) {
 	const promises = [];
+
 	C.DEFAULT_JOBS.forEach(job => {
-		promises.push(this.addJob(teamspace, job));
+		promises.push(Job.addJob(teamspace, job));
 	});
 
 	return Promise.all(promises);
-
 };
 
-schema.statics.usersWithJob = function(teamspace) {
-	return this.find({account: teamspace}, {}, {_id: 1, users : 1}).then((jobs) => {
-		const userToJob  = {};
+Job.addJob = async function(teamspace, jobData) {
+	if (!jobData._id || !validateJobName(jobData._id)) {
+		throw responseCodes.JOB_ID_INVALID;
+	}
 
-		jobs.forEach(job => {
-			job.users.forEach(user => {
-				userToJob[user] = job._id;
-			});
-		});
+	const foundJob = await Job.findByJob(teamspace, jobData._id);
+
+	if (foundJob) {
+		throw responseCodes.DUP_JOB;
+	}
 
-		return userToJob;
+	const jobsColl = await getCollection(teamspace);
+	const newJobEntry = {
+		_id: jobData._id,
+		users: []
+	};
 
-	});
+	if (jobData.color) {
+		newJobEntry.color = jobData.color;
+	}
 
+	return jobsColl.insert(newJobEntry);
 };
 
-schema.statics.removeUserFromAnyJob = function(teamspace, user) {
-	return Job.findByUser(teamspace, user).then(job => {
-		if(job) {
-			return job.removeUserFromJob(user);
-		}
-	});
+Job.addUserToJob = async function(teamspace, jobName, user) {
+	// Check if user is member of teamspace
+	const User = require("./user");
+	await User.teamspaceMemberCheck(teamspace, user);
+
+	const job = await Job.findByJob(teamspace, jobName);
+
+	if (!job) {
+		return Promise.reject(responseCodes.JOB_NOT_FOUND);
+	}
 
+	await Job.removeUserFromAnyJob(teamspace, user);
+
+	job.users.push(user);
+
+	const jobsColl = await getCollection(teamspace);
+	return jobsColl.update({_id: jobName}, {$set: {users: job.users}});
 };
 
-schema.methods.removeUserFromJob = function(user) {
-	this.users.splice(this.users.indexOf(user), 1);
-	return this.save();
+Job.findByJob = async function(teamspace, jobName) {
+	const jobsColl = await getCollection(teamspace);
+	const foundJob = await jobsColl.findOne({_id: jobName});
+
+	if (foundJob && !foundJob.users) {
+		foundJob.users = [];
+	}
+
+	return foundJob;
 };
 
-schema.statics.findUsersWithJobs = function (teamspace, jobs) {
-	return this.find({ account: teamspace }, { _id: { $in: jobs } })
-		.then(items => items.reduce((users, jobitem) => users.concat(jobitem.users),[]));
+Job.findByUser = async function(teamspace, user) {
+	const jobsColl = await getCollection(teamspace);
+	const foundJob = await jobsColl.findOne({users: user});
+
+	if (foundJob && !foundJob.users) {
+		foundJob.users = [];
+	}
+
+	return foundJob;
 };
 
-schema.statics.findByJob = function(teamspace, job) {
-	return this.findOne({account: teamspace}, {_id: job});
+Job.findUsersWithJobs = async function(teamspace, jobNames) {
+	const jobsColl = await getCollection(teamspace);
+	const foundJobs = await (await jobsColl.find({ _id: { $in: jobNames } })).toArray();
```

Comment:
you don't need an await on the find. It returns a cursor. It only returns a promise when you call .toArray()

https://docs.mongodb.com/drivers/node/usage-examples/find